content/kotlin/concepts/hashmap/hashmap.md

```diff
@@ -0,0 +1,145 @@
+---
+Title: 'HashMap'
+Description: 'A HashMap is used to store items as a key-value pairs. The keys and values can be of either same or different types.'
+Subjects:
+  - 'Computer Science'
+Tags:
+  - 'HashMap'
+  - 'Data Types'
+  - 'Collections'
+  - 'Data Structures'
+CatalogContent:
+  - 'learn-kotlin'
+  - 'paths/computer-science'
+---
+
+A `HashMap` is a data structure that maps keys to values and is implemented using a hash table. It is an efficient way to store and retrieve data because it uses a hash function to map keys to indices in an array, so that keys can be found quickly. In Kotlin, a `HashMap` is represented by the `HashMap` class.
+
+## Syntax
+
+```pseudo
+val map = HashMap<KeyType, ValueType>()
+```
+
+Here, `KeyType` is the type of the keys in the map, and `ValueType` is the type of the values.
+
+The types can also be specified when creating an instance of the `HashMap` class like this:
+
+```pseudo
+val map: HashMap<KeyType, ValueType> = HashMap()
+```
+
+The `mutableMapOf()` function is used to create a mutable HashMap:
+
+```pseudo
+val map = mutableMapOf<KeyType, ValueType>()
+```
+
+The `mapOf()` function is used to create an immutable `HashMap`:
+
+```pseudo
+val map = mapOf<KeyType, ValueType>()
+```
```

Comment:
I don't have much experience with Kotlin, but I had a few questions about these lines.

- Isn't the `mutableMapOf()` more closely related to `MutableMap` (which is the interface for `HashMap`)?
- Doesn't the `mapOf()` have more to do with the `Map` interface?

These questions came to mind after doing a little research through [the official Kotlin docs](https://kotlinlang.org/docs/home.html).

Reply:
Damm I'm impressed that you asked this question just by reading the docs from the site. I usually learn stuff from Coursera. But I'm trying to read the docs too. Specially the jetbrains doc for kotlin is amazing. 
- Yes, you are right. The `mutableMapOf()` function in Kotlin creates an instance of a `MutableMap`, which is an interface that is implemented by the HashMap class. This function is a shorthand for creating a new `HashMap` and populating it with key-value pairs.
- I think it is not directly related to the Map interface in, although it can be used to create a map object that implements the Map interface.